src/Core/Services/UserService.cs

```diff
@@ -167,6 +167,19 @@ public async Task<Organization> GetOrganizationAsync(string id)
             }
             return new Organization(organizations[id]);
         }
+        
+        public async Task<Organization> GetOrganizationByIdentifierAsync(string identifier)
+        {
+            var userId = await GetUserIdAsync();
+            var organizations = await GetAllOrganizationAsync();
+
+            if (organizations == null || organizations.Count == 0)
+            {
+                return null;
+            }
+
+            return organizations.Find(o => o.Identifier == identifier);
```

Comment:
Would `.FirstOrDefault()` be more appropriate here?

Reply:
Yes, indeed. TS -> C# failed syntax translations :D